app/models/service_template_ansible_playbook.rb

```diff
@@ -24,13 +24,40 @@ def self.default_retirement_entry_point
   #     :reconfigure (same as provision)
   #
   def self.create_catalog_item(options, _auth_user)
+    task_id = create_catalog_item_queue(options, _auth_user)
+    task = MiqTask.wait_for_taskid(task_id)
+    task.task_results
+  end
+
+  def self.create_catalog_item_queue(options, auth_user)
+    task_opts = {
+      :action => "Create Ansible Playbook Service Template",
+      :userid => "system"
+    }
+
+    playbook = ManageIQ::Providers::AnsibleTower::AutomationManager::Playbook.find(options.fetch_path(:config_info, :provision, :playbook_id))
+    tower = playbook.manager
+
+    queue_opts = {
+      :args        => [options, auth_user],
+      :class_name  => "ServiceTemplateAnsiblePlaybook",
+      :method_name => "create_catalog_item_task",
+      :priority    => MiqQueue::HIGH_PRIORITY,
+      :role        => "ems_operations",
+      :zone        => tower.my_zone
+    }
+
+    MiqTask.generic_action_with_callback(task_opts, queue_opts)
+  end
+
+  def self.create_catalog_item_task(options, _auth_user)
     options      = options.merge(:service_type => 'atomic', :prov_type => 'generic_ansible_playbook')
     service_name = options[:name]
     description  = options[:description]
     config_info  = validate_config_info(options)
 
     transaction do
-      create(options.except(:config_info)).tap do |service_template|
+      create(options.except(:config_info, :task_id)).tap do |service_template|
```

Comment:
With #13822 merged, now you can do `create_from_options(options.except(:task_id)).tap`